src/cdk/overlay/dispatchers/overlay-keyboard-dispatcher.ts

```diff
@@ -28,7 +28,9 @@ export class OverlayKeyboardDispatcher extends BaseOverlayDispatcher {
 
     // Lazily start dispatcher once first overlay is added
     if (!this._isAttached) {
-      this._document.body.addEventListener('keydown', this._keydownListener);
+      this.ngZone.runOutsideAngular(() =>
```

Comment:
It may be better to do this `BaseOverlayDispatcher` so that other dispatchers can take advantage of it. Also while we're at it, should we have a similar change in `OverlayOutsideClickDispatcher`?

Reply:
Hey, I've done the same within the `OverlayOutsideClickDispatcher`, not sure how to do that within the `BaseOverlayDispatcher`. Do you have any ideas?

Reply:
My thinking was that we could move some more logic into the base class, but looking at it again it seems like it would only be the `ngZone.runOutSideAngular` call. Let's leave it as it is.